learn_script.py

```diff
@@ -61,61 +58,82 @@ def run_command(content, free=False, auto=False):
     return True
 
 
-processing = ''
-tmp_processing = ''
-
-for line in script:
-    if processing:
-        if line.strip() == "###":
-            open(processing, 'w').write(tmp_processing)
-            processing = ''
-            tmp_processing = []
-        tmp_processing += line
-        continue
-
-    if not line.strip():
-        continue
-
-    line = line.strip()
-
-    if ' ' not in line:
-        line += ' '
-
-    command, content = line.split(' ', 1)
-    content = content.strip()
-
-    if command == '!checkfile':  # checkfile not exists
-        if not checkfile(content, False):
-            exit(1)
-    if command == 'checkfile':  # checkfile
-        if not checkfile(content):
-            exit(1)
-    if command == '##':
-        ansiprint(content)
-        help_txt.append(content)
-    if command == '#':
-        ansiprint(content)
-    if command == 'enter':
-        enter()
-    if command == 'run_free':
-        if not run_command(content, free=True):
-            exit(1)
-    if command == 'run':
-        if not run_command(content):
-            exit(1)
-    if command == 'run_auto':
-        if not run_command(content, auto=True):
-            exit(1)
-    if command == '###':
-        processing = content
+def process(script):
+
+    processing = ''
+    tmp_processing = ''
+
+    for line in script:
+        if processing:
+            if line.strip() == "###":
+                open(processing, 'w').write(tmp_processing)
+                processing = ''
+                tmp_processing = []
+            tmp_processing += line
+            continue
+
+        if not line.strip():
+            continue
+
+        line = line.strip()
+
+        if ' ' not in line:
+            line += ' '
+
+        command, content = line.split(' ', 1)
+        content = content.strip()
+
+        if command == '!checkfile':  # checkfile not exists
+            if not checkfile(content, False):
+                exit(1)
+        if command == 'checkfile':  # checkfile
+            if not checkfile(content):
+                exit(1)
+        if command == '##':
+            ansiprint(content)
+            help_txt.append(content)
+        if command == '#':
+            ansiprint(content)
+        if command == 'enter':
+            enter()
+        if command == 'run_free':
+            if not run_command(content, free=True):
+                exit(1)
+        if command == 'run':
+            if not run_command(content):
+                exit(1)
+        if command == 'run_auto':
+            if not run_command(content, auto=True):
+                exit(1)
+        if command == '###':
+            processing = content
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Run Tutorterminal application.")
+
+    parser.add_argument('files', metavar='files', type=str, nargs='+',
+                        help='the script files to run')
+
+    args = parser.parse_args()
+    num_scripts = len(args.files)
```

Comment:
validate files before calling:

```suggestion
    invalid_files = list(filter(lambda x: x[0], map(lambda x: (not os.path.isfile(x), x), args.files)))
    if invalid_files:
        for _, fname in invalid_files:
             print(fname + " not found")
    num_scripts = len(args.files)
```

Reply:
Nice!